parsers/mysql/mysql.go

```diff
@@ -81,16 +98,83 @@ type SlowQuery struct {
 	LockTime        float64   `json:"lock_time"`
 	RowsSent        int       `json:"rows_sent"`
 	RowsExamined    int       `json:"rows_examined"`
-	Query           string    `json:"query",omitempty`
+	Query           string    `json:"query,omitempty"`
 	NormalizedQuery string    `json:"normalized_query,omitempty"`
+	DB              string    `json:"db,omitempty"`
+	HostedOn        string    `json:"hosted_on,omitempty"`
+	ReadOnly        bool      `json:"read_only,omitempty"`
 	skipQuery       bool
 }
 
-func (p *Parser) Init(_ interface{}) error {
+func (p *Parser) Init(options interface{}) error {
+	p.conf = *options.(*Options)
 	p.nower = &RealNower{}
+	if p.conf.Host != "" {
+		url := fmt.Sprintf("%s:%s@tcp(%s)/", p.conf.User, p.conf.Pass, p.conf.Host)
+		db, err := sql.Open("mysql", url)
+		if err != nil {
+			log.Fatal(err)
+		}
+		p.db = db
+	}
+
+	// update hostedOn and readOnly every <n> seconds
+	go func() {
+		ticker := time.NewTicker(time.Second * time.Duration(p.conf.QueryInterval))
+		for _ = range ticker.C {
+			p.hostedOn = getHostedOn(p.db)
+			p.readOnly = getReadOnly(p.db)
+		}
+	}()
+
 	return nil
 }
 
+func getReadOnly(db *sql.DB) bool {
+	rows, err := db.Query("SELECT @@global.read_only;")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer rows.Close()
+
+	var value bool
+
+	rows.Next()
+	if err := rows.Scan(&value); err != nil {
+		log.Fatal(err)
+	}
+	if err = rows.Err(); err != nil {
```

Comment:
probably want err := rows.Err() here since it's within the if ... ; err {} style.